manuscript/16-CompFound.txt

```diff
@@ -0,0 +1,756 @@
+
+Δημιουργούνται διάφορες δομές δεδομένων για να ανταποκρίνονται σε συγκεκριμένες
+εφαρμογές, όπως στοίβες, ουρές, δέντρα και γραφήματα. Οι μηχανικοί λογισμικού
+ενθαρρύνονται να μάθουν τις διελεύσεις μέσω μη γραμμικών δομών δεδομένων,
+οι οποίες περιλαμβάνουν διαφορετικούς αναλυτές δέντρων (διελεύσεις δέντρων
+προ-διάταξης, εντός-διάταξης και μετά-διάταξης), λειτουργίες CRUD σε δέντρα,
+εξισορρόπηση δέντρων, δυαδικά δέντρα αναζήτησης (BST), δέντρα AVL και δέντρα
+κόκκινου-μαύρου, και να μάθουν αλγορίθμους αναζήτησης δέντρων (πρώτα βάθος,
+πρώτα πλάτος, συντομότερες διαδρομές κ.λπ.). Μερικά από αυτά συζητούνται
+στο γνωστικό κλάδο Μαθηματικά Θεμέλια.
+
+
+
+### Αλγόριθμοι και χαρακτηριστικά των αλγορίθμων
+
+
+
+Όλα τα λογισμικά υλοποιούν λογική για την εκτέλεση της απαιτούμενης λειτουργίας.
+Αυτή η λογική ή ο αλγόριθμος για την εκτέλεση μιας συγκεκριμένης εργασίας πρέπει
+να σχεδιάζεται ή να επιλέγεται με γνώμονα την απόδοση του συστήματος, την ασφάλεια,
+τη φορητότητα, τη συντηρησιμότητα, την επεκτασιμότητα και την απλότητα, μεταξύ άλλων. 
+
+Η πολυπλοκότητα ενός αλγορίθμου προσδιορίζεται με τη μέτρηση των υπολογιστικών
+πόρων (υπολογιστική ισχύς και χώρος) που καταναλώνει ο εν λόγω αλγόριθμος για
+ένα δεδομένο σύνολο δεδομένων. 
+
+Η βαθιά κατανόηση των δομών δεδομένων είναι ζωτικής σημασίας για την ανάλυση και
+το σχεδιασμό καλών αλγορίθμων. Ανατρέξτε στην περιοχή περιεχομένου "Δομές δεδομένων
+και οργάνωση" για περισσότερες λεπτομέρειες. 
+
+Τα χαρακτηριστικά των αλγορίθμων είναι πολλά και περιλαμβάνουν τη λειτουργικότητα,
+την ορθότητα, την ευρωστία, την αρθρωτότητα, τη συντηρησιμότητα, τη φιλικότητα προς
+τον προγραμματιστή (ευκολία ενσωμάτωσης στο έργο και ευκολία χρήσης), τη φιλικότητα
+προς τον χρήστη (δηλαδή πόσο εύκολα γίνεται κατανοητός από τους ανθρώπους), την
+ανάγκη για χρόνο του προγραμματιστή, την απλότητα και την επεκτασιμότητα. 
+
+Ένα συνήθως τονισμένο χαρακτηριστικό των αλγορίθμων είναι η "απόδοση" ή η
+"αποδοτικότητα". 
+
+Οι παράμετροι που έχουν σημασία για την κατανάλωση πόρων ενός αλγορίθμου
+περιλαμβάνουν, μεταξύ άλλων, τα εξής:
+
+1. Υλικό 
+2. Λογισμικό 
+3. Επιλογή και σχεδιασμός αλγορίθμου για συγκεκριμένο πρόβλημα 
+4. Αποτελεσματική υλοποίηση
+
+
+
+### Πολυπλοκότητα αλγορίθμου
+
+
+
+Η πολυπλοκότητα ενός αλγορίθμου είναι ένα μέτρο των πόρων που καταναλώνει
+(υπολογιστική ισχύς ή μνήμη) για ένα συγκεκριμένο πρόβλημα και δεδομένο
+σύνολο δεδομένων. 
+
+Η επιλογή των σωστών δομών δεδομένων και των πράξεων σε δομές δεδομένων
+και η εξασφάλιση της βέλτιστης υλοποίησης του αλγορίθμου επηρεάζουν
+επίσης την πολυπλοκότητα του αλγορίθμου.
+
+
+
+### Μέτρηση της πολυπλοκότητας
+
+
+
+Συχνά, η πολυπλοκότητα ενός αλγορίθμου δηλώνεται από τους πόρους που καταναλώνονται
+στο χειρότερο σενάριο. Η πολυπλοκότητα των αλγορίθμων μετράται συνήθως με ασυμπτωτικούς
+συμβολισμούς για τα σενάρια καλύτερης περίπτωσης, χειρότερης περίπτωσης και μέσης
+περίπτωσης όσον αφορά την κατανάλωση πόρων για ένα δεδομένο σύνολο δεδομένων. 
```

Comment:
(e.g., sorted, unsorted, and sorted in
reverse order),  δε βρίσκω αυτό το κομμάτι

Reply:
@TeoSkondras η μετάφραση βρίσκεται στις γραμμές 598-600.